src/rust/engine/process_execution/src/local.rs

```diff
@@ -114,6 +115,79 @@ impl CommandRunner {
   }
 }
 
+lazy_static! {
+  static ref IS_LIKELY_IN_DOCKER: bool = is_likely_in_docker().unwrap_or_else(|e| {
+    log::info!("Failed to detect whether we are running in docker: {}", e);
+    false
+  });
+}
+
+///
+/// Attempts to detect whether we are running inside a docker container.
+///
+/// NB: Do not call this directly: it is stored in the IS_LIKELY_IN_DOCKER lazy_static.
+///
+fn is_likely_in_docker() -> Result<bool, String> {
```

Comment:
Shouldn't this check include a check for /bin/sync since its all moot unless that magic executable exists?

Reply:
It actually looks like sync is not guaranteed to be synchronous, its just guaranteed to synchrnously _schedule_ a sync: https://linux.die.net/man/8/sync

Reply:
> IOW - we seem to have evidence docker / aufs is not the problem.

Before landing this change, I confirmed using the repro from #10557 that it fixes the issue: see Eric's comment above.

I rebased it out, but could put it back in.